utils/http/interfaces.go

```diff
@@ -2,17 +2,32 @@
  * Copyright (C) 2020-2022 Arm Limited or its affiliates and Contributors. All rights reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
+
+// Package http provides a familiar HTTP client interface with
+// various implementations in order to answer common use cases: e.g. need for a robust client (performs retries on server errors with different retry policies depending on configuration),
+// for a fast client which leverages connection pools.
+// For the robust client, it is possible to set its configuration fully but this package also come with a set of preset configuration to answer most usecases
+// e.g. to create a client which performs exponential backoff and listens to `Retry-After` headers, the following can be done:
```

Comment:
Will these show as examples in godoc? For reference I found this: https://pkg.go.dev/github.com/fluhus/godoc-tricks (https://github.com/fluhus/godoc-tricks)

Reply:
I followed this https://github.com/fluhus/godoc-tricks/blob/1266307515d621e91d7972ddf41b003c6838219f/doc.go#L117

Reply:
I added the extra indent as suggested